UIforETW/Utility.cpp

```diff
@@ -352,75 +743,112 @@ int DeleteFiles(HWND hwnd, const std::vector<std::wstring>& paths)
 		FOF_ALLOWUNDO | FOF_FILESONLY | FOF_NOCONFIRMATION,
 	};
 	// Delete using the recycle bin.
-	int result = SHFileOperation(&fileOp);
+	//TODO: IFileOperation?
+	const int result = ::SHFileOperationW(&fileOp);
 
 	return result;
 }
 
 void SetClipboardText(const std::wstring& text)
 {
-	BOOL cb = OpenClipboard(GetDesktopWindow());
-	if (!cb)
+	if (!openClipboard())
 		return;
 
-	EmptyClipboard();
+	if (!emptyClipboard())
+	{
+		closeClipboard();
+		return;
+	}
 
-	size_t length = (text.size() + 1) * sizeof(wchar_t);
-	HANDLE hmem = GlobalAlloc(GMEM_MOVEABLE, length);
-	if (hmem)
+	HANDLE hmem_temp = FALSE;
+	if (!AllocGlobalMemoryForString(text, &hmem_temp))
 	{
-		void *ptr = GlobalLock(hmem);
-		if (ptr != NULL)
-		{
-			memcpy(ptr, text.c_str(), length);
-			GlobalUnlock(hmem);
+		closeClipboard();
+		return;
+	}
+	const HANDLE& hmem = hmem_temp;
 
-			SetClipboardData(CF_UNICODETEXT, hmem);
-		}
+	void* ptr_temp = nullptr;
+	if (!LockGlobalMemory(hmem, &ptr_temp))
+	{
+		FreeGlobalMemory(hmem);
+		closeClipboard();
+		return;
 	}
 
-	CloseClipboard();
+	void* const ptr = ptr_temp;
+
+	wcscpy_s(static_cast<wchar_t*>(ptr), (text.size() + 1), text.c_str());
+
+	UnlockGlobalMemory(hmem);
+
+	if (!ClipboardSetUnicode(hmem))
+	{
+		FreeGlobalMemory(hmem);
+		closeClipboard();
+		return;
+	}
+	closeClipboard();
 }
 
 std::wstring GetClipboardText()
 {
 	std::wstring result;
-	BOOL cb = OpenClipboard(GetDesktopWindow());
-	if (!cb)
+	if (!openClipboard())
 		return result;
 
-	HANDLE hClip = GetClipboardData(CF_UNICODETEXT);
-	if (hClip)
+	HANDLE hClip_temp = NULL;
+	if (!ClipboardGetUnicode(&hClip_temp))
 	{
-		wchar_t *text = static_cast<wchar_t *>(GlobalLock(hClip));
-		if (text)
-		{
-			size_t bytes = GlobalSize(hClip);
-			result.insert(result.begin(), text, text + bytes / sizeof(wchar_t));
-		}
-		::GlobalUnlock(hClip);
+		closeClipboard();
+		return result;
 	}
 
-	CloseClipboard();
+	const HANDLE hClip = hClip_temp;
 
+	void* ptr = nullptr;
+	if (!LockGlobalMemory(hClip, &ptr))
+	{
+		closeClipboard();
+		return result;
+	}
+
+	PCWSTR const text = static_cast<PCWSTR>(ptr);
+
+	size_t bytes_temp = 0u;
+	if (!SizeGlobalMemory(hClip, &bytes_temp))
+	{
+		UnlockGlobalMemory(hClip);
+		closeClipboard();
+		return result;
+	}
+
+	const size_t bytes = bytes_temp;

```

Comment:
Why copy the value? const variables are nice, but only having one copy of a value is also nice.